platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/internal/IsolatedProjectsSafeKotlinDslScriptsModelBuilder.kt

```diff
@@ -0,0 +1,301 @@
+        exceptions.asSequence().runtimeFailuresLocatedIn(scriptFile.path).map(::exceptionToString).toList()
+
+    private
+    fun exceptionToString(exception: Exception) = exception.stackTraceToString()
+}
+
+
+private
+fun initScriptModels(rootProject: ProjectInternal): List<IntermediateScriptModel> {
+    return rootProject.discoverInitScripts().map {
+        initScriptModelBuilder(it, rootProject).buildModel()
+    }
+}
+
+
+private
+fun settingsScriptModels(rootProject: ProjectInternal): IntermediateScriptModel? {
+    return rootProject.discoverSettingScript()?.let {
+        settingsScriptModelBuilder(it, rootProject).buildModel()
+    }
+}
+
+
+private
+fun initScriptModelBuilder(scriptFile: File, project: ProjectInternal) = project.run {
+
+    val (scriptHandler, scriptClassPath) = compilationClassPathForScriptPluginOf(
+        target = gradle,
+        scriptFile = scriptFile,
+        baseScope = gradle.classLoaderScope,
+        scriptHandlerFactory = scriptHandlerFactoryOf(gradle),
+        project = project,
+        resourceDescription = "initialization script"
+    )
+
+    IntermediateScriptModelBuilder(
+        scriptFile = scriptFile,
+        project = project,
+        scriptClassPath = scriptClassPath,
+        scriptSourcePath = sourcePathFor(listOf(scriptHandler))
+    )
+}
+
+
+private
+fun settingsScriptModelBuilder(scriptFile: File, project: Project) = project.run {
+
+    IntermediateScriptModelBuilder(
+        scriptFile = scriptFile,
+        project = project,
+        scriptClassPath = settings.scriptCompilationClassPath,
+        scriptSourcePath = sourcePathFor(listOf(settings.buildscript))
+    )
+}
+
+
+internal
+data class IntermediateScriptModel(
+    val scriptFile: File,
+    val classPath: ClassPath,
+    val sourcePath: ClassPath,
+    val implicitImports: List<String>,
+)
+
+
+private
+data class IntermediateScriptModelBuilder(
+    val scriptFile: File,
+    val project: Project,
+    val scriptClassPath: ClassPath,
+    val accessorsClassPath: (ClassPath) -> AccessorsClassPath = { AccessorsClassPath.empty },
+    val scriptSourcePath: ClassPath,
+    val additionalImports: () -> List<String> = { emptyList() }
+) {
+
+    fun buildModel(): IntermediateScriptModel {
+        val classPathModeExceptionCollector = project.serviceOf<ClassPathModeExceptionCollector>()
+        val accessorsClassPath =
+            classPathModeExceptionCollector.ignoringErrors {
+                accessorsClassPath(scriptClassPath)
+            } ?: AccessorsClassPath.empty
+
+        val additionalImports =
+            classPathModeExceptionCollector.ignoringErrors {
+                additionalImports()
+            } ?: emptyList()
+
+        val classPath = (scriptClassPath + accessorsClassPath.bin)
+        val sourcePath = (gradleSource() + scriptSourcePath + accessorsClassPath.src)
+        val implicitImports = project.scriptImplicitImports + additionalImports
```

Comment:
🤔 I tried a distribution built from this branch and after a sync, IDEA behaves as if the project build scripts don't have an implicit import for `org.gradle.kotlin.dsl.*`

Reply:
Ah, I see. After a small change, editing simple build scripts works again:
```diff
--- a/platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/internal/Isola
tedProjectsSafeKotlinDslScriptsModelBuilder.kt
+++ b/platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/internal/Isola
tedProjectsSafeKotlinDslScriptsModelBuilder.kt
@@ -96,9 +96,9 @@ class IsolatedProjectsSafeKotlinDslScriptsModelBuilder(
     private
     fun buildModel(midModel: IntermediateScriptModel, exceptions: List<Exception>): KotlinDslScriptModel {
         return StandardKotlinDslScriptModel(
-            classPath = emptyList(),
-            sourcePath = emptyList(),
-            implicitImports = emptyList(),
+            classPath = midModel.classPath.asFiles,
+            sourcePath = midModel.sourcePath.asFiles,
+            implicitImports = midModel.implicitImports,
             editorReports = emptyList(),
             exceptions = getExceptionsForFile(midModel.scriptFile, exceptions)
         )
```